src/init.cpp

```diff
@@ -800,7 +800,7 @@ std::set<BlockFilterType> g_enabled_filter_types;
     std::terminate();
 };
 
-bool AppInitBasicSetup(const ArgsManager& args)
+bool AppInitBasicSetup(const ArgsManager& args, NodeContext& node)
```

Comment:
In commit "refactor: index: use `AbortNode` in fatal error helper" (3c06926cf21dcca3074ef51506f556b2286c299b)

It would be slightly better to just pass a reference to node.exit_status instead of passing the whole NodeContext, because passing NodeContext exposes a lot of other state this function doesn't need to have access to. But it's not important because #27711 gets rid of the InitShutdownState function and will drop the parameter anyway.

Reply:
done. Refactored to pass `exit_status` ref.